### ISSUE TITLE
Allow for default methods to still be called over HTTP

### ISSUE THREAD

**Issue by billbindi** [OpenFeign/feign#615]
If we have the `default` implementation for an interface, it would be nice to have a way to specify that it _should_ still be called over HTTP. For example say we have the following interface (stolen form the readme):
```java
interface GitHub {
  @RequestLine("GET /users/{username}/repos?sort={sort}")
  List<Repo> repos(@Param("username") String owner, @Param("sort") String sort);

  @RequestLine("GET /users/{username}/repos")
  default List<Repo> repos(String owner) {
    return repos(owner, "full_name");
  }
}
```
We want there to be a way to call either version of the `repos` method over the wire, and if a sort is not input then it just uses the default of "full_name." There may be a way to get around this with default values on `@Param` (we use `@RequestParam`, actually, which does allow a default), or maybe checking for `null` and falling back to a default, but if we want the second `repos` (the one with only one param) to do a bit more work (e.g. write to a log that the method was called without a sort, or choose a default based on config settings, then it would be more complicated.

To ground this more realistically, what we actually care about is adding to an existing API interface without making a breaking change. That is, going from:
```java
@Path("/foo")
interface Foo {
  @Path("/bar")
  void bar();
}
```
to
```java
@Path("/foo")
interface Foo {
  @Path("/bar")
  void bar();

  @Path("/baz")
  default void baz() {
    throw new UnsupportedOperationException("Baz is not supported right now.");
  }
}
```
is not a breaking change for any implementer of the `Foo` API. Here, we care about having a default implementation to save it from breaking consumers, but we also want to have feign call a real implementation on a remote server.

One suggestion for this is to allow for an `@IgnoreDefault` annotation (or some other name) that feign checks in `ReflectiveFeign#newInstance` when checking if the method is a default method to short circuit and not care about the existence of the `default` keyword. Then, it will get a normal handler that still tries to make the remote call.

**Comment by kdavisk6** [comment]
My gut response to this is that it sounds like this is something that might be outside the scope of Feign.  However, I will mark this as a question for us to discuss.

**Comment by rage-shadowman** [comment]
@billbindi The example in the readme does not have any annotations on the default method. Not sure if that makes a difference.

**Comment by kdavisk6** [comment]
Would limiting which methods are proxied to only those annotated with be useful?  That is a more incremental change we can try without the need to add additional annotations and update the contract.  Thoughts?

**Comment by billbindi** [comment]
Yes, that would be perfect. It is not required that _all_ methods are proxied, but having some way to single out specific methods would be nice.

**Comment by kdavisk6** [comment]
Closed in favor of #686
